### PR TITLE
JAMES-3874 LMTP should handle overQuota exceptions

### PR DESCRIPTION
**Pull request by Maxxx873**

### CODE REVIEW

File: server/protocols/protocols-lmtp/src/main/java/org/apache/james/lmtpserver/hook/MailboxDeliverToRecipientHandler.java [hunk 1]
```diff
@@ -105,6 +105,13 @@ public long size() {
                 .smtpReturnCode(SMTPRetCode.MAIL_OK)
                 .smtpDescription(DSNStatus.getStatus(DSNStatus.SUCCESS, DSNStatus.CONTENT_OTHER) + " Message received <" + recipient.asString() + ">")
                 .build();
+        } catch (OverQuotaException e) {
+            LOGGER.error("OverQuota error handling ", e);
```
Comment (chibenwa) [inline server/protocols/protocols-lmtp/src/main/java/org/apache/james/lmtpserver/hook/MailboxDeliverToRecipientHandler.java:109]: Imo this is eventually expected upon regular operation, INFO  is likely ok.

"{} is overquota" is likely a more infirmative message. Pass target username as a parameter too to maximize usefullness.

File: server/protocols/protocols-lmtp/src/test/java/org/apache/james/lmtpserver/LmtpServerTest.java [hunk 2]
```diff
@@ -73,15 +80,16 @@
...
 import org.apache.mailet.Mail;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Nested;
-import org.junit.jupiter.api.Test;
 
 import com.github.fge.lambdas.Throwing;
 import com.google.inject.name.Names;
 
+import org.junit.jupiter.api.AfterEach;
```
Comment (chibenwa) [inline server/protocols/protocols-lmtp/src/test/java/org/apache/james/lmtpserver/LmtpServerTest.java:87]: Careful to import order changes: you likely break checkstyles.

**Review by chibenwa — COMMENTED** [review]
Nice little contib!
Thanks!

**Review by Arsnael — COMMENTED** [review]
Read it, nothing to add. After the little fixes asked by @chibenwa , it should be approved without issues I think.
Thanks for the contribution :)

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**